docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,623 @@
+
+The `Query._entities` field returns a list of objects that belong to the `_Entity` union type, which is why this union must include _all_ entities that the subgraph defines.
+
+#### Example
+
+Let's say a supergraph includes two subgraphs with the following schemas:
+
+<CodeColumns>
+
+```graphql title="Products"
+type Product @key(fields: "upc") {
+  upc: String!
+  name: String!
+}
+
+type Query {
+  topProducts: [Product!]!
+}
+```
+
+
+```graphql title="Reviews"
+type Product @key(fields: "upc") {
+  upc: String!
+  reviews: [Review!]!
+}
+
+type Review {
+  score: Int!
+  description: String!
+}
+```
+
+</CodeColumns>
+
+With these subgraph schemas, a client can execute the following query against the graph router:
+
+```graphql
+query GetTopProductReviews {
+  topProducts {
+    reviews {
+      description
+    }
+  }
+}
+```
+
+To resolve this query, the graph router _starts_ by sending the following query to the Products subgraph, because that's where the top-level `Query.topProducts` field is defined:
+
+```graphql
+query {
+  topProducts {
+    __typename
+    upc
+  }
+}
+```
+
+Notice that this query includes `Product.__typename` and `Product.upc`, even though those fields _aren't_ included in the original client query. The graph router knows that these two fields are used in a `Product` type's **representation**, which it will use to fetch the remaining fields from the Reviews subgraph.
+
+After getting this result from the Products subgraph, the router can send this followup query to the Reviews subgraph:
+
+```graphql
+query ($_representations: [_Any!]!) {
+  _entities(representations: $_representations) {
+    ... on Product {
+      reviews {
+        description
+      }
+    }
+  }
+}
+```
+
+Notice that this query uses inline fragment matching (`... on Product`), because the return type of `Query._entities` is the `_Entity` union type.
+
+Each entry that the router includes in the `$_representations` list variable has the following shape:
+
+```json
+{
+  "__typename": "Product",
+  "upc": "B00005N5PF"
+}
+```
+
+These are the representation fields that the router automatically added to its Products query.
+
+### Resolving `Query._entities`
+
+As a reminder, here's the definition of the `Query._entities` field that every subgraph must automatically define (unless that subgraph defines zero entities):
+
+```graphql {2}
+type Query {
+  _entities(representations: [_Any!]!): [_Entity]!
+}
+```
+
+Every subgraph must also automatically define the resolver for this field. The  logic for this resolver is as follows:
+
+1. For each entity representation included in the `representations` list, obtain the `__typename` from that representation.
+2. Execute the **reference resolver** for the corresponding `__typename` and provide the full representation to that resolver.
```

Comment:
I suppose a "reference resolver" is how `@apollo/subgraph` happens to handle resolving representations, but it isn't the mandatory solution for other libraries, right? So I think we need to make this section a little more implementation-agnostic

Reply:
yes, libraries have to implement this logic